### DISCUSSION TITLE
请教下，在bfe中怎么识别不同的请求协议的？

### DISCUSSION THREAD

**Discussion by zyong** [bfenetworks/bfe#855] [Q&A (问题和回答)] [answered]
如题，就是bfe中怎么区别不同协议的请求的，是通过port还是分析协议head ，我在想如果是http和https可以通过port区分，到http2和quic我理解是不是只能分析特征来区别，然后bfe还支持其他协议，总体的思路是怎么样的呢？

**Comment by mileszhang2016** [comment, marked as answer]
1. http、https是基于端口来区分的，这两个协议的底层都是TCP，标准端口是80和443；
2. quic的底层是UDP（以避免TCP的Head of Line Blocking问题），有独立的协议端口。目前BFE开源项目还没有提供对quic的支持。
3. https和http/2都基于TLS（https其实是http over TLS），它们使用相同的端口（这个端口是TLS使用的），通过TLS的协商机制来决定所使用的协议。

可以参考wikepedia上的说明
https://en.wikipedia.org/wiki/HTTP/2
